File: DIDemo1/app/src/main/java/com/androidtutz/anushka/didemo/view/MainActivity.java
package com.androidtutz.anushka.didemo.view;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.androidtutz.anushka.didemo.App;
import com.androidtutz.anushka.didemo.R;
import com.androidtutz.anushka.didemo.model.SmartPhone;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {

    @Inject
    SmartPhone smartPhone;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ((App) getApplication()).getComponent().inject(this);
        smartPhone.makeACall();

        findViewById(R.id.button)
                .setOnClickListener(
                        view -> {
                            startActivity(new Intent(this, Main2Activity.class));
                            finish();
                        }
                );
    }
}

File: ViewModelDemo1UpgradeLesson/app/src/main/java/com/anushka/viewmodeldemo1/view/MainActivity.kt
package com.anushka.viewmodeldemo1.view

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.anushka.viewmodeldemo1.R
import com.anushka.viewmodeldemo1.databinding.ActivityMainBinding
import com.anushka.viewmodeldemo1.viewmodel.MainActivityViewModel
import com.anushka.viewmodeldemo1.viewmodel.factory.MainViewModelFactory

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var viewModel: MainActivityViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = DataBindingUtil.setContentView(this,R.layout.activity_main)
        viewModel = ViewModelProvider(this, MainViewModelFactory(5)).get(MainActivityViewModel::class.java)

        setUpView()
        setUpObservers()
    }

    private fun setUpObservers() {
        viewModel.getSum().observe(this, Observer {
            binding.resultSum.text = it.toString()
        })
    }

    private fun setUpView() {
        binding.button.setOnClickListener {
            viewModel.makeSum(binding.inputNumber.text.toString().toInt())
        }
    }
}
